java/src/main/java/io/cucumber/java/JavaDefaultDataTableCapitalCaseTransformerDefinition.java

```diff
@@ -0,0 +1,77 @@
+package io.cucumber.java;
+
+import io.cucumber.core.backend.Lookup;
+import io.cucumber.core.runtime.Invoker;
+import io.cucumber.datatable.TableCellByTypeTransformer;
+import io.cucumber.datatable.TableEntryByTypeTransformer;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class JavaDefaultDataTableCapitalCaseTransformerDefinition extends JavaDefaultDataTableEntryTransformerDefinition {
+    JavaDefaultDataTableCapitalCaseTransformerDefinition(Method method, Lookup lookup) {
+        super(method, lookup);
+    }
+
+    @Override
+    public TableEntryByTypeTransformer tableEntryByTypeTransformer() {
+        return super.tableEntryByTypeTransformer();
+    }
+
+    @Override
+    Object execute(Map<String, String> fromValue, Type toValueType, TableCellByTypeTransformer cellTransformer) throws Throwable {
+        Object[] args;
+        if (method.getParameterTypes().length == 3) {
+            args = new Object[]{fromValue, toValueType, cellTransformer};
+        } else {
+            args = new Object[]{fromValue, toValueType};
+        }
+        args[0] = convertToCamelCase((Map<String, Object>)args[0]);
```

Comment:
I suggested earlier to merge these classes. But had that not been the case you could reduce the duplication here a bit as follows:

You've duplicated this method to modify `arg[0]`. However `arg[0]` will always be `fromValue`. So you can modify `fromValue` and then called `super.execute` with the modified `fromValue`.